src/constraints.jl

```diff
@@ -1020,7 +1025,8 @@ function _constraint_primal(
         },
     },
     result::Int,
-)::Float64
+    ::Type{T},
+)::T where {T}
```

Comment:
This is a bug: https://github.com/jump-dev/JuMP.jl/issues/3380

Reply:
Fixed in https://github.com/jump-dev/JuMP.jl/pull/3381. We should merge that first, and then rebase this PR on top.